eslint/babel-eslint-plugin/src/rules/semi.js

```diff
@@ -76,30 +76,38 @@ function report(context, node, missing) {
   });
 }
 
+function checkForSemicolon(node, context) {
+  const options = context.options[1];
+  const exceptOneLine = options && options.omitLastInOneLineBlock === true;
+
+  const sourceCode = context.getSourceCode();
+  const lastToken = sourceCode.getLastToken(node);
+
+  if (context.options[0] === "never") {
+    if (isUnnecessarySemicolon(context, lastToken)) {
+      report(context, node, true);
+    }
+  } else {
+    if (!isSemicolon(lastToken)) {
+      if (!exceptOneLine || !isOneLinerBlock(context, node)) {
+        report(context, node);
+      }
+    } else {
+      if (exceptOneLine && isOneLinerBlock(context, node)) {
+        report(context, node, true);
+      }
+    }
+  }
+}
+
 export default ruleComposer.joinReports([
   rule,
   context => ({
     ClassProperty(node) {
```

Comment:
@kaicataldo Does `matches` [selector](https://eslint.org/docs/developer-guide/selectors) support
```
":matches(ClassProperty, ClassPrivateProperty)"
```
?

Reply:
Yes! As an example, you can see this core rule: https://github.com/eslint/eslint/blob/master/lib/rules/func-call-spacing.js#L171

We should be able to combine both of these selectors into one, e.g. `"ClassProperty, ClassPrivateProperty"(node) {}`.

Reply:
Good to know that `A, B` is supported, guess we should update https://eslint.org/docs/developer-guide/selectors#what-syntax-can-selectors-have.